lisa/operating_system.py

```diff
@@ -131,16 +170,47 @@ def _get_detect_string(cls, node: Any) -> Iterable[str]:
         cmd_result = typed_node.execute(cmd="cat /etc/SuSE-release", no_error_log=True)
         yield get_matched_str(cmd_result.stdout, cls.__suse_release_pattern)
 
+    def _get_os_version(self) -> OsVersion:
+        return self._os_version
+
 
 class Windows(OperatingSystem):
     def __init__(self, node: Any) -> None:
         super().__init__(node, is_posix=False)
+        self._get_os_version
+
+    def _get_os_version(self) -> OsVersion:
+        self._os_version.vendor = "Microsoft Corporation"
+        cmd_result = self._node.execute(
+            cmd="systeminfo | findstr /B /C:'OS Version'",
+            no_error_log=True,
+        )
+        if cmd_result.exit_code == 0 and cmd_result.stdout != "":
+            for row in cmd_result.stdout.split("\n"):
+                if ": " in row:
+                    key, value = row.split(": ")
+                    if "OS Version" in key:
+                        self._os_version.release = value.strip()
+        return self._os_version
 
 
 class Posix(OperatingSystem, BaseClassMixin):
+
+    __url_pattern = re.compile(
+        r"^(?:http|ftp)s?://"  # http:// or https://
+        r"(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)"
```

Comment:
Can we just check `startswith`? It looks enough to me to check if it's an url.

Reply:
Hmm ok we can do that for now. if there arises some case where we need more, we can go back to this regex.